tests/conftest.py

```diff
@@ -316,6 +317,24 @@ def netrc_other_host(monkeypatch: pytest.MonkeyPatch, tmp_path: Path) -> Path:
     return netrc_file
 
 
+@pytest.fixture
+def netrc_home_directory(monkeypatch: pytest.MonkeyPatch, tmp_path: Path) -> Path:
+    """Create a netrc file in a mocked home directory without setting NETRC env var."""
+    # Create a fake home directory with .netrc or _netrc (Windows)
+    home_dir = tmp_path / "home"
+    home_dir.mkdir()
+    netrc_filename = "_netrc" if platform.system() == "Windows" else ".netrc"
+    netrc_file = home_dir / netrc_filename
+    netrc_file.write_text("default login netrc_user password netrc_pass\n")
+
+    # Mock Path.home() to return our fake home directory
+    monkeypatch.setattr("pathlib.Path.home", lambda: home_dir)
```

Comment:
I think you should be able to use

```suggestion
    monkeypatch.setenv('HOME', str(home_dir))
```

right?

Reply:
I think we need to use `USERPROFILE` on windows